// Copyright 2015 gRPC authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.analise";
option java_outer_classname = "MensageiroProto";
option objc_class_prefix = "Mens";

package mensageiro;

// The greeting service definition.
service Mensageiro {
  // Sends a greeting
  rpc EnviarMensagemVazia (Resposta_Request_Void) returns (Resposta_Request_Void) {}
  rpc EnviarMensagemLong (Resposta_Request_Long) returns (Resposta_Request_Long) {}
  rpc EnviarMensagemOitoLong (Resposta_Request_Long) returns (Resposta_Request_Long) {}
  rpc EnviarMensagemString (Resposta_Request_String) returns (Resposta_Request_String) {}  
}

// The request message containing the user's name.
message Resposta_Request_Void {}

// The response message containing the greetings
message Resposta_Request_Long {
  float message = 1;
}

message Resposta_Request_String {
  string message = 1;
}


